src/clj/rems/db/organizations.clj

```diff
@@ -0,0 +1,59 @@
+(ns rems.db.organizations
+  (:require [clojure.core.memoize :as memo]
+            [medley.core :refer [update-existing]]
+            [rems.api.schema :refer [OrganizationFull OrganizationOverview]]
+            [rems.common.util :refer [index-by]]
+            [rems.db.core :as db]
+            [rems.db.roles :as roles]
+            [rems.json :as json]
+            [schema.coerce :as coerce])
+  (:import [org.joda.time DateTime]))
+
+(def ^:private +organizations-cache-time-ms+ (* 5 60 1000))
+
+(defn add-organization! [userid org]
+  (db/add-organization! {:id (:organization/id org)
+                         :user userid
+                         :time (DateTime.)
+                         :data (json/generate-string org)})
+  {:success true
+   :organization/id (:organization/id org)})
+
+(def ^:private coerce-organization-overview
+  (coerce/coercer! OrganizationOverview json/coercion-matcher))
+
+(def ^:private coerce-organization-full
+  (coerce/coercer! OrganizationFull json/coercion-matcher))
+
+(defn- parse-organization [raw]
+  (merge
+   (json/parse-string (:data raw))
+   {:organization/modifier {:userid (:modifieruserid raw)}
+    :organization/last-modified (:modified raw)}))
+
+(defn get-organizations-raw []
+  (->> (db/get-organizations)
+       (map parse-organization)
+       (map coerce-organization-full)))
+
+;; TODO unify caching behavior and location https://github.com/CSCfi/rems/issues/2179
+(def ^:private organizations-by-id-cache (memo/ttl #(index-by [:organization/id] (get-organizations-raw)) :ttl/threshold +organizations-cache-time-ms+))
```

Comment:
I don't see a cache reset for these. I'd maybe prefer to merge this feature first without caching, unless that's an unreasonable performance hit.

Reply:
The intention was to make it expire by itself but yes this may not do it when there is 24/7 use. I will add the missing `get-organization-by-id` and use that instead. The caching concern is in #2179 so let's do it then.

Reply:
:+1:

Reply:
Fixed